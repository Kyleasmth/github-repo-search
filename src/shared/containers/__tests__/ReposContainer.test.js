import React from 'react';
import RepoContainer from '../ReposContainer';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';


jest.mock('react-table/react-table.css', () => null);


describe('RepoContainer group Test', () => {
  let instance;
  const props = {
    repos : [
      {id: 123, name: 'abc'},
      {id: 543, name: 'xync'},
      {id: 123, name: 'rsc'},
    ],
    setCache: jest.fn()
  };
  beforeEach(() => {
    instance = shallow(<RepoContainer.reactComponent {...props} />).instance();
  });
  it('RepoContainer: snapshot test', () => {
    const snapshot = renderer.create(<RepoContainer.reactComponent repos={[]} setCache={jest.fn} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('componentDidMount: should call setCache and set repos', () => {
    expect(instance.props.setCache).toHaveBeenCalled();
    expect(instance.state.repos).toEqual(props.repos);
  });
  it('toggleHeader: should sort column', () => {
    const selectedHeader = {Header: 'Repo Title', accessor: 'name', sortBy: 'name'};
    const sortedRepos = [
      {'id': 123, 'name': 'abc'}, 
      {'id': 123, 'name': 'rsc'}, 
      {'id': 543, 'name': 'xync'}
    ];
    expect(instance.state.selectedHeader).toEqual({});
    instance.toggleHeader(selectedHeader)();
    expect(instance.state.selectedHeader).toEqual(selectedHeader);
    expect(instance.state.headerValue).toEqual(false);
    expect(instance.state.repos).toEqual(sortedRepos);
  });
  it('onNextPress: should increment startIndex', () => {
    instance.onNextPress();
    expect(instance.state.startIndex).toBe(5);
    instance.onNextPress();
    expect(instance.state.startIndex).toBe(10);
  });
  it('onPreviousPress: should decrement startIndex', () => {
    instance.setState({startIndex: 25});
    instance.onPreviousPress();
    expect(instance.state.startIndex).toBe(20);
    instance.onPreviousPress();
    expect(instance.state.startIndex).toBe(15);
  });
  it('changeRowsCount: should change rowsCount and reset startIndex', () => {
    instance.setState({startIndex: 25});
    instance.changeRowsCount(10);
    expect(instance.state.rowsCount).toBe(10);
    expect(instance.state.startIndex).toBe(0);
  });
});