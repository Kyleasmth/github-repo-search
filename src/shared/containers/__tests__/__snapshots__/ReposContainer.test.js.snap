// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoContainer group Test RepoContainer: snapshot test 1`] = `
<div
  className="sc-ifAKCX dMxYvZ"
>
  <div
    className="ReactTable sc-bxivhb dtKShY"
    style={Object {}}
  >
    <div
      className="rt-table"
      role="grid"
      style={undefined}
    >
      <div
        className="rt-thead -header"
        style={
          Object {
            "minWidth": "500px",
          }
        }
      >
        <div
          className="rt-tr"
          role="row"
          style={undefined}
        >
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              ID
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Repo Title
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Owner
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Stars
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Created
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="rt-tbody"
        style={
          Object {
            "minWidth": "500px",
          }
        }
      >
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="rt-noData"
    >
      No rows found
    </div>
    <div
      className="-loading"
    >
      <div
        className="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
  <div
    className="sc-bwzfXH hbVBfZ"
  >
    <button
      className="waves-effect waves-light btn-small sc-htpNat iKiwZL"
      disabled={true}
      onClick={[Function]}
    >
      Previous
    </button>
    <div
      className="input-field"
    >
      <select
        className="sc-bdVaJa kZAgjJ"
        onChange={[Function]}
      >
        <option
          value={5}
        >
          Rows to Display
           
          5
        </option>
        <option
          value={10}
        >
          Rows to Display
           
          10
        </option>
        <option
          value={15}
        >
          Rows to Display
           
          15
        </option>
        <option
          value={20}
        >
          Rows to Display
           
          20
        </option>
      </select>
    </div>
    <button
      className="waves-effect waves-light btn-small sc-htpNat iKiwZL"
      disabled={true}
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
